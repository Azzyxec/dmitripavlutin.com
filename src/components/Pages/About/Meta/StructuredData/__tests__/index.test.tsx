import { shallow } from 'enzyme';

import AboutMetaStructuredData from '../index';

const siteInfo = {
  title: 'Dmitri Pavlutin',
  description: 'Thoughts on Frontend development',
  metaTitle: 'Dmitri Pavlutin Blog',
  metaDescription: 'Posts by Dmitri Pavlutin about software development',
  url: 'https://dmitripavlutin.com',
  repositoryUrl: 'https://github.com/panzerdp/dmitripavlutin.com',
};

const authorInfo = {
  name: 'Dmitri Pavlutin',
  description: 'Dmitri Pavlutin is a software developer specialized in Frontend technologies',
  email: 'mail@mail.com',
  jobTitle: 'Software developer',
  profiles: {
    stackoverflow: 'https://stackoverflow.com/users/1894471/dmitri-pavlutin',
    twitter: 'https://twitter.com/panzerdp',
    linkedin: 'https://www.linkedin.com/in/dmitri-pavlutin/',
    github: 'https://github.com/panzerdp',
    facebook: 'https://www.facebook.com/dmitri.pavlutin',
  },
  nicknames: {
    twitter: 'panzerdp',
  },
};

const authorProfilePictureSrc = '/profile.png';

const props = {
  siteInfo,
  authorInfo,
  authorProfilePictureSrc,
};

describe('<AboutMetaStructuredData />', function() {
  it('should render structured data', function() {
    const wrapper = shallow(<AboutMetaStructuredData {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
