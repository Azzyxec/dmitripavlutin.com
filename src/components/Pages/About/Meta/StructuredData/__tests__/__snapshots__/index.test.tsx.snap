// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutMetaStructuredData /> should render structured data 1`] = `
<HelmetWrapper
  defer={true}
  encodeSpecialCharacters={true}
>
  <script
    type="application/ld+json"
  >
    {
  "@context": "https://schema.org",
  "@type": "Person",
  "name": "Dmitri Pavlutin",
  "image": {
    "@type": "ImageObject",
    "url": "https://dmitripavlutin.com/profile.png",
    "width": 256,
    "height": 256
  },
  "url": "https://dmitripavlutin.com",
  "sameAs": [
    "https://stackoverflow.com/users/1894471/dmitri-pavlutin",
    "https://twitter.com/panzerdp",
    "https://www.linkedin.com/in/dmitri-pavlutin/",
    "https://github.com/panzerdp",
    "https://www.facebook.com/dmitri.pavlutin"
  ],
  "description": "Dmitri Pavlutin is a software developer specialized in Frontend technologies",
  "jobTitle": "Software developer",
  "email": "mail@mail.com"
}
  </script>
</HelmetWrapper>
`;
